pkg/pxc/pxc.go

```diff
@@ -21,27 +21,31 @@ const appName = "pxc"
 
 var NoProxyDetectedError = errors.New("can't detect enabled proxy, please enable HAProxy or ProxySQL")
 
-func GetPrimaryPodDNSName(ctx context.Context, cl client.Client, cr *api.PerconaXtraDBCluster) (string, error) {
-	primary, err := GetPrimaryPod(ctx, cl, cr)
-	if err != nil {
-		return "", errors.Wrap(err, "get primary pod")
-	}
-	pxcSet := statefulset.NewNode(cr)
-	podList := corev1.PodList{}
-	if err := cl.List(ctx, &podList, &client.ListOptions{
+func getFirstReadyPodFQDN(ctx context.Context, cl client.Client, cr *api.PerconaXtraDBCluster) (string, error) {
```

Comment:
The behaviour of this function is that it actually waits for ALL the pods to be ready and then get the [0], but IMO the name implies that it gets any ready pod.